src/coprocessor/dag/expr/builtin_string.rs

```diff
@@ -37,6 +39,31 @@ impl ScalarFunc {
         let i = try_opt!(self.children[0].eval_int(ctx, row));
         Ok(Some(Cow::Owned(format!("{:b}", i).into_bytes())))
     }
+
+    #[inline]
+    pub fn ltrim<'a, 'b: 'a>(
+        &'b self,
+        ctx: &mut EvalContext,
+        row: &'a [Datum],
+    ) -> Result<Option<Cow<'a, [u8]>>> {
+        let val = try_opt!(self.children[0].eval_string(ctx, row));
```

Comment:
Call `eval_string_and_decode` is better than combine `eval_string` and `String::from_utf8`.

Reply:
I have tested it with MySQL. Seems we don't need parse raw bytes to string here. Just trim the byte slice is OK.
```sql
create table x (a binary(32));
insert into x values (x'ffff');
select hex(trim(a)) from x
+------------------------------------------------------------------+
| hex(trim(a))                                                     |
+------------------------------------------------------------------+
| FFFF000000000000000000000000000000000000000000000000000000000000 |
+------------------------------------------------------------------+
1 row in set (0.00 sec)
```

Reply:
Same behavior with `TiDB`.

Reply:
@hicqu  I've  refined the implementation here. PTAL :)